pip/req/req_set.py

```diff
@@ -204,6 +204,32 @@ def locate_files(self):
                     (req_to_install, req_to_install.source_dir)
                 )
 
+    @staticmethod
+    def _available_extras(provided, requested):
+        """
+        Given a tuple of installable extras, and a list of extras that the user
```

Comment:
Oh oops, just saw that @xavfernandez suggested removing the static methods and simply using `sets` so this wouldn't be applicable anymore.